extension/js/content_scripts/webmail/gmail_element_replacer.ts

```diff
@@ -323,21 +325,29 @@ export class GmailElementReplacer implements WebmailElementReplacer {
     return nRenderedAtts;
   }
 
-  private filterAtts = (potentialMatches: JQueryEl | HTMLElement, patterns: string[]) => {
+  private renderBackupFromFile = async (attMeta: Att, attsContainerInner: JQueryEl, msgEl: JQueryEl, isOutgoing: boolean, attSel: JQueryEl, nRenderedAtts: number) => {
+    let downloadedAtt: R.GmailAtt;
+    try {
+      downloadedAtt = await Google.gmail.attGet(this.acctEmail, attMeta.msgId!, attMeta.id!); // .id! is present when fetched from api
+    } catch (e) {
+      attsContainerInner.show().addClass('attachment_processed').find('.attachment_loader').text('Please reload page');
+      nRenderedAtts++;
+      return nRenderedAtts;
+    }
+    const openpgpType = await BrowserMsg.send.bg.await.pgpMsgType({ rawBytesStr: downloadedAtt.data.toRawBytesStr() });
```

Comment:
Here you are doing type detection of the content based on the first few bytes of the attachment.

This is useful for public keys received from outside, because our regex to detect them is quite vague, and sometimes we're not sure if it's pubkey or not from the name alone.

But for backups (private keys), the regex is very specific, so it's unlikely any normal file will accidentally end up being wrongly treated as a backup. You can skip this detection and just treat the file as a backup in all cases.